sdk/tracing/tracing-opentelemetry/src/main/java/com/azure/tracing/opentelemetry/implementation/AmqpPropagationFormatUtil.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.tracing.opentelemetry.implementation;
+
+import com.azure.core.implementation.tracing.Tracer;
+import com.azure.core.util.Context;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.TraceOptions;
+import io.opencensus.trace.Tracestate;
+
+
+public class AmqpPropagationFormatUtil {
+    private static final String SPAN_CONTEXT = Tracer.OPENTELEMETRY_AMQP_EVENT_SPAN_CONTEXT;
+
+    private AmqpPropagationFormatUtil() { }
+
+    /**
+     * This method is called to extract the Span Context information from the received event's diagnostic Id.
+     *
+     * @param diagnosticId The diagnostic Id for the event.
+     * @return {@link Context}
+     */
+    public static Context extractContext(String diagnosticId) {
+        return new Context(SPAN_CONTEXT, fromDiagnosticId(diagnosticId));
+    }
+
+    /**
+     * Parse span context and get the diagnostic Id for that event.
+     *
+     * @param spanContext the span context.
+     * @retrun The dignostic Id
+     */
+    public static String getDiagnosticId(SpanContext spanContext) {
+        char[] chars = new char[55];
```

Comment:
Is there a spec to look at for this?

Reply:
This is the spec - https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#file-minimal_instrumentation-md

Reply:
Can we add documentation in the javadocs how this gets created? A year from now, someone else looking at this would have no idea.